test/integration_tests/src/test_latency_aware_policy.cpp

```diff
@@ -0,0 +1,245 @@
+#endif
+
+    test_utils::CassFuturePtr close_future(cass_session_close(session_.get()));
+    cass_future_wait(close_future.get());
+  }
+
+  /**
+   * Start the query execution thread
+   */
+  void start_query_execution() {
+    uv_thread_create(&thread_, start_thread, NULL);
+  }
+
+  /**
+   * Stop the executing query thread
+   */
+  void stop_query_execution() {
+    is_running_ = false;
+    uv_thread_join(&thread_);
+  }
+
+  /**
+   * Create latency on a node using CCM to pause/resume execution of the node
+   *
+   * @param node Node to create latency on
+   * @param latency Latency to incur
+   */
+  void create_latency(int node, unsigned int latency) {
+    // Add latency to the node
+    ccm_->pause(node);
+    boost::this_thread::sleep_for(boost::chrono::milliseconds(latency));
+    ccm_->resume(node);
+  }
+
+  /**
+   * Check the maximum latency incurred on a node and ensure that is it within
+   * accepted threshold/tolerance
+   *
+   * @param node Node to check against
+   * @param expected_latency Latency incurred
+   * @param tolerance Acceptable percentage threshold/tolerance
+   */
+  void check_max_latency(int node, unsigned int expected_latency, double tolerance) {
+      double latency = max_node_latency[node - 1];
+      if (latency < expected_latency) {
+        BOOST_CHECK_CLOSE(latency, expected_latency, tolerance);
+      }
+  }
+
+private:
+  static test_utils::CassSessionPtr session_;
+  const cql::cql_ccm_bridge_configuration_t& configuration_;
+  test_utils::CassClusterPtr cluster_;
+  static bool is_running_;
+#if UV_VERSION_MAJOR == 0
+  static uv_loop_t* loop_;
+#else
+  static uv_loop_t loop_;
+#endif
+  uv_thread_t thread_;
+  static double max_node_latency[3];
+
+  /**
+   * Start the executing query thread
+   *
+   * @param arg Not used
+   */
+  static void start_thread(void* arg) {
+    execute_query();
+#if UV_VERSION_MAJOR == 0
+    uv_run(loop_, UV_RUN_DEFAULT)
+#else
+    uv_run(&loop_, UV_RUN_DEFAULT);
+#endif
+  }
+
+  /**
+   * Execute a query on the system table
+   */
+  static void execute_query() {
+    is_running_ = true;
+    while(is_running_) {
+      std::string query = "SELECT * FROM system.local";
+      test_utils::CassStatementPtr statement(cass_statement_new_n(query.data(), query.size(), 0));
+      test_utils::CassFuturePtr future(cass_session_execute(session_.get(), statement.get()));
+      CassError error_code = test_utils::wait_and_return_error(future.get(),
+                                                               240 * test_utils::ONE_SECOND_IN_MICROS);
+
+      // Ignore all timing errors
+      if (error_code != CASS_OK &&
+          error_code != CASS_ERROR_LIB_REQUEST_TIMED_OUT &&
+          error_code != CASS_ERROR_SERVER_READ_TIMEOUT) {
+        CassString message;
+        cass_future_error_message(future.get(), &message.data, &message.length);
+        BOOST_FAIL(std::string(message.data, message.length) << "' (" << cass_error_desc(error_code) << ")");
```

Comment:
I'm weary of calling into Boost test from a thread that's not the main thread. It might be ok in this case because it's the only thread, but i'm not sure. It might have problems if this fails (throws). An alternative solution would be to print to stderr,  mark a failed flag and break the loop. The failed flag can be checked in the main thread.